### PR TITLE
feat: Present proof - signs the presentation

### PR DESCRIPTION
**Pull request by soluchok**
By default, we will sign the presentation with `BbsBlsSignature2020` (if a signature is needed). Also, it can be changed by providing an option e.g `WithAddProofFn`. 

Signed-off-by: Andrii Soluk <isoluchok@gmail.com>

### CODE REVIEW

File: pkg/didcomm/protocol/middleware/presentproof/middlewares.go [hunk 1]
```diff
@@ -103,11 +114,73 @@ type presentationExchangePayload struct {
...
+		if err != nil {
+			return err
+		}
+
+		kid, pubKey, err := km.CreateAndExportPubKeyBytes(kms.BLS12381G2Type)
+		if err != nil {
+			return err
+		}
+
+		methodID := fingerprint.KeyFingerprint(bls12381g2pub, pubKey)
```
Comment (sudeshrshetty) [inline pkg/didcomm/protocol/middleware/presentproof/middlewares.go:156]: @Baha-sk is going to add Create DID Key feature for bls12381 keys, please add a follow up task for him to replace this code with create bbs DID key api.

**Review by sudeshrshetty — APPROVED** [review]

File: pkg/didcomm/protocol/middleware/presentproof/middlewares.go [hunk 2]
```diff
@@ -103,11 +114,73 @@ type presentationExchangePayload struct {
...
+		didKey := fmt.Sprintf("%s#%s", fmt.Sprintf("did:key:%s", methodID), methodID)
+
+		presentation.Context = append(presentation.Context, bbsContext)
+
+		return presentation.AddLinkedDataProof(&verifiable.LinkedDataProofContext{
+			SignatureType:           "BbsBlsSignature2020",
+			SignatureRepresentation: verifiable.SignatureProofValue,
+			Suite: bbsblssignature2020.New(
+				suite.WithSigner(newBBSSigner(km, cr, kid)),
+				suite.WithVerifier(bbsblssignature2020.NewG2PublicKeyVerifier()),
```
Comment (llorllale) [inline pkg/didcomm/protocol/middleware/presentproof/middlewares.go:166]: nit - you don't need a verifier when signing

**Review by llorllale — APPROVED** [review]

File: pkg/didcomm/protocol/middleware/presentproof/middlewares.go (hunk 2 above)
Comment (soluchok) [reply pkg/didcomm/protocol/middleware/presentproof/middlewares.go:166]: I don't think so. I will try to remove it.

File: pkg/didcomm/protocol/middleware/presentproof/middlewares.go (hunk 1 above)
Comment (soluchok) [reply pkg/didcomm/protocol/middleware/presentproof/middlewares.go:156]: Done https://github.com/hyperledger/aries-framework-go/pull/2601

File: pkg/didcomm/protocol/middleware/presentproof/middlewares.go (hunk 2 above)
Comment (soluchok) [reply pkg/didcomm/protocol/middleware/presentproof/middlewares.go:166]: Done https://github.com/hyperledger/aries-framework-go/pull/2601

**Pull request merged**